File: examples/bevholt/bevholt.cpp
#include <TMB.hpp>

template<class Type>
Type objective_function<Type>::operator() ()
{
  DATA_VECTOR(R);
  DATA_VECTOR(S);
  int n = R.size();

  PARAMETER(logRmax);
  PARAMETER(logS50);
  PARAMETER(logSigma);
  Type Rmax = exp(logRmax);
  Type S50 = exp(logS50);
  Type sigma = exp(logSigma);
  vector<Type> Rhat(n);

  Type neglogL = 0;

  Rhat = Rmax * S / (S + S50);
  ADREPORT(Rhat);
  neglogL = -sum(dnorm(R, Rhat, sigma, true));

  return neglogL;
}

File: examples/bevholt/bevholt.R
data <- read.table("bevholt.dat", header=TRUE)
## plot(R~S, data, xlim=c(0,400), ylim=c(0,200))

parameters <- list(logRmax=log(400), logS50=log(100), logSigma=0)

require(TMB)
compile("bevholt.cpp")
dyn.load(dynlib("bevholt"))

################################################################################

model <- MakeADFun(data, parameters, DLL="bevholt")
fit <- nlminb(model$par, model$fn, model$gr)

best <- model$env$last.par.best
rep <- sdreport(model)

print(best)
print(rep)
summary(rep)

Svec <- seq(0, 4000, 20)
Rmax <- exp(best[["logRmax"]])
S50 <- exp(best[["logS50"]])
Rvec <- Rmax * Svec / (Svec + S50)
## lines(Rvec~Svec)
